core/common/src/main/java/alluxio/conf/AlluxioProperties.java

```diff
@@ -0,0 +1,232 @@
+    mSources.clear();
+  }
+
+  /**
+   * Puts the key value pair specified by users.
+   *
+   * @param key key to put
+   * @param value value to put
+   */
+  public void put(String key, String value) {
+    mUserProps.put(key, Optional.ofNullable(value));
+  }
+
+  /**
+   * Remove the value set to key.
+   *
+   * @param key key to put
+   */
+  public void remove(String key) {
+    mUserProps.put(key, Optional.empty());
+  }
+
+  /**
+   * Checks if there is a value set for the given key.
+   *
+   * @param key the key to check
+   * @return true if there is value for the key, false otherwise
+   */
+  public boolean hasValueSet(String key) {
+    if (mUserProps.containsKey(key)) {
+      return mUserProps.get(key).isPresent();
+    }
+    return PropertyKey.isValid(key) && (PropertyKey.fromString(key).getDefaultValue() != null);
+  }
+
+  /**
+   * @return the entry set of all Alluxio property key and value pairs (value can be null)
+   */
+  public Set<Map.Entry<String, String>> entrySet() {
+    Set<Map.Entry<String, String>> entrySet =
+        Stream.concat(
+            PropertyKey.defaultKeys().stream()
+                .filter(key -> !mUserProps.containsKey(key.getName()))
+                .map(key -> Maps.immutableEntry(key.getName(), key.getDefaultValue())),
+            mUserProps.entrySet().stream()
+                .map(entry -> Maps.immutableEntry(entry.getKey(), entry.getValue().orElse(null))))
+            .collect(toSet());
+
+    return entrySet;
+  }
+
+  /**
+   * @return the entry set of all Alluxio property key and value pairs (value can be null)
+   */
+  public Set<String> keySet() {
+    Set<String> keySet =
+        PropertyKey.defaultKeys().stream().map(PropertyKey::getName).collect(toSet());
+    keySet.addAll(mUserProps.keySet());
+    return keySet;
+  }
+
+  /**
+   * Iterates over all the key value pairs and performs the given action.
+   *
+   * @param action the operation to perform on each key value pair
+   */
+  public void forEach(BiConsumer<? super String, ? super String> action) {
+    for (Map.Entry<String, String> entry : entrySet()) {
+      action.accept(entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * Sets the source for a given key.
+   *
+   * @param key property key
+   * @param source the source
+   */
+  public void setSource(String key, Source source) {
+    mSources.put(key, source);
+  }
+
+  /**
+   * @param key property key
+   * @return the source of the key
+   */
+  public Source getSource(String key) {
+    Source source = mSources.get(key);
+    if (source != null) {
+      return source;
+    }
+    if (PropertyKey.isValid(key)) {
+      return Source.DEFAULT;
+    }
+    return Source.UNKNOWN;
+  }
+
+  /**
+   * Merges the current configuration properties with alternate properties. A property from the new
+   * configuration wins if it also appears in the current configuration.
+   *
+   * @param properties The source {@link Properties} to be merged
```

Comment:
The -> the 
here and on the line below to be consistent within this file

Reply:
fixed